clients/include/testing_trmm.hpp

```diff
@@ -27,138 +28,177 @@ void testing_trmm(const Arguments& arg)
         }
 
         if(arg.unit_check)
         {
-            unit_check_general<T>(M, N, ldb, hB_copy, hC);
+            if(std::is_same<T, rocblas_half>{} && K > 10000)
+            {
+                // For large K, rocblas_half tends to diverge proportional to K
+                // Tolerance is slightly greater than 1 / 1024.0
+                const double tol = K * sum_error_tolerance<T>;
+                near_check_general<T>(M, N, ldb, cpuB, hB_1, tol);
+                near_check_general<T>(M, N, ldb, cpuB, hB_2, tol);
+            }
+            else
+            {
+                unit_check_general<T>(M, N, ldb, cpuB, hB_1);
+                unit_check_general<T>(M, N, ldb, cpuB, hB_2);
+            }
         }
 
         if(arg.norm_check)
         {
-            rocblas_error = norm_check_general<T>('F', M, N, ldb, hB_copy, hC);
+            auto err1     = std::abs(norm_check_general<T>('F', M, N, ldb, cpuB, hB_1));
+            auto err2     = std::abs(norm_check_general<T>('F', M, N, ldb, cpuB, hB_2));
+            rocblas_error = err1 > err2 ? err1 : err2;
+            rocblas_error = err1;
         }
     }
 
     if(arg.timing)
     {
-        // only norm_check return an norm error, unit check won't return anything
-        cout << "M, N, lda, rocblas-Gflops (us) ";
-        if(arg.norm_check)
+        int number_cold_calls = 2;
+        int number_hot_calls  = arg.iters;
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+
+        for(int i = 0; i < number_cold_calls; i++)
         {
-            cout << "CPU-Gflops(us), norm-error";
+            CHECK_ROCBLAS_ERROR(rocblas_trmm<T>(
+                handle, side, uplo, transA, diag, M, N, &alpha_h, dA, lda, dB, ldb));
         }
-        cout << endl;
-
-        cout << M << ',' << N << ',' << lda << ',' << rocblas_gflops << "(" << gpu_time_used
-             << "),";
 
-        if(arg.norm_check)
+        gpu_time_used = get_time_us(); // in microseconds
+        for(int i = 0; i < number_hot_calls; i++)
         {
-            cout << cblas_gflops << "(" << cpu_time_used << "),";
-            cout << rocblas_error;
+            rocblas_trmm<T>(handle, side, uplo, transA, diag, M, N, &alpha_h, dA, lda, dB, ldb);
         }
+        gpu_time_used  = get_time_us() - gpu_time_used;
+        rocblas_gflops = trmm_gflop_count<T>(M, N, side) * number_hot_calls / gpu_time_used * 1e6;
+
+        std::cout << "M,N,alpha,lda,ldb,side,uplo,transA,diag,rocblas-Gflops,us";
+
+        if(arg.unit_check || arg.norm_check)
+            std::cout << ",CPU-Gflops,us,norm-error";
+
+        std::cout << std::endl;
+
+        std::cout << M << ',' << N << ',' << arg.get_alpha<T>() << ',' << lda << ',' << ldb << ','
+                  << char_side << ',' << char_uplo << ',' << char_transA << ',' << char_diag << ','
+                  << rocblas_gflops << "," << gpu_time_used;
```

Comment:
You must divide gpu_time_used by number of hot calls for accurate logging.

Reply:
Thanks for catching this, fixed in commit 80858ac